model_hub/model_hub/huggingface/_arg_parser.py

```diff
@@ -0,0 +1,279 @@
+import dataclasses
```

Comment:
Can we refer to this as "config parsing" rather than "arg parsing"?  I'm concerned that the name suggests that it is for command-line arguments, the way that `ArgParse` is, but it seems like it's really dedicated to parsing hparam dictionaries and the like